js/app/modules/contentGroup/mediaLightbox.js

```diff
@@ -66,44 +73,80 @@ const MediaLightbox = new Module.Class({
+            this._preview_media_object(resource);
             this._loading_new_lightbox = false;
-            let media_object;
-            try {
-                media_object = engine.get_object_by_id_finish(task);
-            } catch (error) {
-                logError(error);
-                return;
-            }
+        } else {
+            Engine.get_default().get_object_by_id(resource, null, (engine, task) => {
+                this._loading_new_lightbox = false;
+                let media_object;
+                try {
+                    media_object = engine.get_object_by_id_finish(task);
+                } catch (error) {
+                    logError(error);
+                    return;
+                }
 
-            // If the next object is not the last, the forward arrow should be displayed.
-            this._preview_media_object(media_object);
-            this._loading_new_lightbox = false;
-        });
+                // If the next object is not the last, the forward arrow should be displayed.
+                this._preview_media_object(media_object);
+                this._loading_new_lightbox = false;
+            });
+        }
     },
 
     _preview_media_object: function (media_object) {
-        if (this._article_model === null)
+        if (this._context === null)
             return;
-        let resources = this._article_model.resources;
-        this._current_index = resources.indexOf(media_object.ekn_id);
+        let resources = this._context;
+
+        this._current_index = resources.map((item) => {
+            if (item instanceof ContentObjectModel.ContentObjectModel) {
+                return item.ekn_id;
+            }
+            return item;
+        }).indexOf(media_object.ekn_id);
+
         if (this._current_index === -1)
             return;
 
         if (this.lightbox_widget)
             this.drop_submodule(this.lightbox_widget);
-        let card = this.create_submodule('card', {
+
+        let widget = this.create_submodule('card', {
             model: media_object
         });
-        this.lightbox_widget = card;
+
+        // Lightboxes may need to show a variety of different content, in which
+        // case a single card type may not fit all needs. If no card type is
+        // specified, try to determine content type and show it accordingly.
+        let content_type = media_object.content_type;
+        if (widget === null) {
+            if (media_object instanceof MediaObjectModel.VideoObjectModel) {
+                widget = new EosKnowledgePrivate.MediaBin();
+                widget.set_uri('file:///home/endless/checkout/eos-knowledge-lib/YUaYs_7vGAI.mp4')
+            } else if (content_type === 'application/pdf') {
+                let stream = media_object.get_content_stream();
+                widget = new PDFView.PDFView({
+                    expand: true,
+                    // FIXME: Is this gross? Any way to make it less gross?
```

Comment:
We do the same thing in `Card.KnowledgeDocument`

https://github.com/endlessm/eos-knowledge-lib/blob/master/js/app/modules/card/knowledgeDocument.js#L218